src/js/darkwire.js

```diff
@@ -105,46 +101,78 @@ export default class Darkwire {
     return importKeysPromises;
   }
 
+  createUser(resolve, reject, username) {
+    Promise.all([
+      this._cryptoUtil.createPrimaryKeys()
+    ])
+    .then((data) => {
+      this._keys = {
+        public: data[0].publicKey,
+        private: data[0].privateKey
+      };
+      return Promise.all([
+        this._cryptoUtil.exportKey(data[0].publicKey, 'spki')
+      ]);
+    })
+    .then((exportedKeys) => {
+      if (!exportedKeys) {
+        return reject('Could not create a user session');
+      }
+
+      return resolve({
+        username: username,
+        publicKey: exportedKeys[0]
+      });
+    });
+  }
+
+  checkSessionUsernames(username) {
+    let matches = _.find(this._users, (users) => {
+      return username.toLowerCase() === users.username.toLowerCase();
+    });
+
+    if (matches && matches.username) {
+      return matches;
+    }
+
+    return false;
+  }
+
   removeUser(data) {
     this._users = _.without(this._users, _.findWhere(this._users, {id: data.id}));
     return this._users;
   }
 
-  updateUsername(username, newUsername) {
+  updateUsername(username) {
```

Comment:
Are we sanitizing usernames? Is it possible to change your name to "<script>alert('hax0r');</script>" ?